pkg/operator/endpoints/metrics.go

```diff
@@ -0,0 +1,449 @@
+/*
```

Comment:
Can you move most of this logic to the `operator` package (e.g. `operator/workloads/metrics.go`)? We've tried to keep the endpoint files thin, basically just for parsing/validating/responding to the request. Maybe calling into that package after line 58 would make sense.